crypto/keys/schnorr/schnorr.go

```diff
@@ -0,0 +1,212 @@
+package schnorr
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"github.com/cometbft/cometbft/crypto"
+	"github.com/cometbft/cometbft/crypto/tmhash"
+	"go.dedis.ch/kyber/v3/sign/schnorr"
+	"go.dedis.ch/kyber/v3/suites"
+	"go.dedis.ch/kyber/v3/util/key"
+
+	errorsmod "cosmossdk.io/errors"
+
+	cryptotypes "github.com/cosmos/cosmos-sdk/crypto/types"
+	"github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// TODO -> Check error handling
+const (
+	PubKeyName  = "tendermint/PubKeySchnorr"
+	PrivKeyName = "tendermint/PrivKeySchnorr"
+	// PubKeySize is is the size, in bytes, of public keys as used in this package.
+	PubKeySize = 32
+	// PrivKeySize is the size, in bytes, of private keys as used in this package.
+	PrivKeySize = 32
+	// SignatureSize Size of an Schnorr signature
+	SignatureSize = 64
+
+	keyType = "schnorr"
+	curve   = "Ed25519"
```

Comment:
Can we name this file sr25519? Schnorr is generalized like Ecdsa

Reply:
This isn't sr25519. sr25519 is

1) A specific challenge function (based on merlin)
2) Ristretto
3) A specific signature encoding

This is the EdDSA Ed25519 challenge function (inherited from the kyber library) applied to an arbitrary curve (in this case specified to Ed25519) with kyber's specific checks for signature validity (it does check a lot of edge cases for Ed25519, yet it doesn't follow zebra and instead defines its own sets of checks will likely produce verification edge cases with any other library).

Reply:
we could use https://github.com/oasisprotocol/curve25519-voi as it does cover zip-215 checks